### ISSUE TITLE
EventDetailURL %file_sha% is using wrong identifier

### ISSUE THREAD

**Issue by jamiestrachan-shopify** [google/santa#795]
Hello,
It appears that when Santa is showing the Blocked Event dialog, it is populating the EventDetailURL with the wrong identifier.

Here is the dialog:
![image](https://user-images.githubusercontent.com/98484497/166483929-fb49fa4c-fc88-43bb-add9-d97e9faa7f6e.png)

And, here is the resulting EventDetailURL (redacted):
```
https://{server path}/8a2c4baecb2c589444f8c85fadae9c166941c52c4f9ea10e647f4bac08037049
```

It appears that the EventDetailURL was populated with the `Bundle Identifier`, and not the (file) `Identifier`.
The application in question is: [Suspicious Package](https://mothersruin.com/software/SuspiciousPackage/)

Output of fileinfo for _Suspicious Package_:
```
santactl fileinfo /Applications/Suspicious\ Package.app/Contents/MacOS/Suspicious\ Package
Path                   : /Applications/Suspicious Package.app/Contents/MacOS/Suspicious Package
SHA-256                : d32bc39e1c387b211b63d50bc6895b26712203bd01e73de413fa5eed78dd4bf8
SHA-1                  : 95b215987d3fd47946f0056d125b93e154e1fd72
Bundle Name            : Suspicious Package
Bundle Version         : 932
Bundle Version Str     : 4.2.1
Team ID                : 936EB786NH
Type                   : Executable (arm64, x86_64)
Code-signed            : Yes
Rule                   : Blocked (Unknown)
Signing Chain:
    1. SHA-256             : 79a15aa105028002ef8f5ad525d4a910496624d570aeac4a7ef5fab45eb9c31a
       SHA-1               : 853556456d17d0203325d92242020f005edb85e7
       Common Name         : Developer ID Application: Randy Saldinger (936EB786NH)
       Organization        : Randy Saldinger
       Organizational Unit : 936EB786NH
       Valid From          : 2019/04/22 10:28:41 -0400
       Valid Until         : 2024/04/22 10:28:41 -0400

    2. SHA-256             : 7afc9d01a62f03a2de9637936d4afe68090d2de18d03f29c88cfb0b1ba63587f
       SHA-1               : 3b166c3b7dc4b751c9fe2afab9135641e388e186
       Common Name         : Developer ID Certification Authority
       Organization        : Apple Inc.
       Organizational Unit : Apple Certification Authority
       Valid From          : 2012/02/01 17:12:15 -0500
       Valid Until         : 2027/02/01 17:12:15 -0500

    3. SHA-256             : b0b1730ecbc7ff4505142c49f1295e6eda6bcaed7e2c68c5be91b5a11001f024
       SHA-1               : 611e5b662c593a08ff58d14ae22452d198df6c60
       Common Name         : Apple Root CA
       Organization        : Apple Inc.
       Organizational Unit : Apple Certification Authority
       Valid From          : 2006/04/25 17:40:36 -0400
       Valid Until         : 2035/02/09 16:40:36 -0500

```

Here is the body of the uploaded event, showing the `file_bundle_hash` matching the generated URL:
```
{
    "events":[
        {
            "ppid": 1,
            "file_bundle_version_string": "4.2.1",
            "signing_chain": [
                {
                    "cn": "Developer ID Application: Randy Saldinger (936EB786NH)",
                    "valid_until": 1713796121,
                    "org": "Randy Saldinger",
                    "valid_from": 1555943321,
                    "ou": "936EB786NH",
                    "sha256": "79a15aa105028002ef8f5ad525d4a910496624d570aeac4a7ef5fab45eb9c31a"
                },
                {
                    "cn": "Developer ID Certification Authority",
                    "valid_until": 1801519935,
                    "org": "Apple Inc.",
                    "valid_from": 1328134335,
                    "ou": "Apple Certification Authority",
                    "sha256": "7afc9d01a62f03a2de9637936d4afe68090d2de18d03f29c88cfb0b1ba63587f"
                },
                {
                    "cn": "Apple Root CA",
                    "valid_until": 2054670036,
                    "org": "Apple Inc.",
                    "valid_from": 1146001236,
                    "ou": "Apple Certification Authority",
                    "sha256": "b0b1730ecbc7ff4505142c49f1295e6eda6bcaed7e2c68c5be91b5a11001f024"
                }
            ],
            "file_bundle_path": "/Applications/Suspicious Package.app",
            "file_name": "Suspicious Package",
            "file_bundle_version": "932",
            "file_bundle_binary_count": 6,
            "parent_name": "launchd",
            "executing_user": "********",
            "file_bundle_name": "Suspicious Package",
            "current_sessions": [ ******** ],
            "quarantine_timestamp": 0,
            "file_bundle_id": "com.mothersruin.SuspiciousPackageApp",
            "file_path": "/Applications/Suspicious Package.app/Contents/MacOS",
            "file_bundle_hash": "8a2c4baecb2c589444f8c85fadae9c166941c52c4f9ea10e647f4bac08037049",
            "file_sha256": "d32bc39e1c387b211b63d50bc6895b26712203bd01e73de413fa5eed78dd4bf8",
            "decision": "BLOCK_UNKNOWN",
            "pid": 35999,
            "file_bundle_executable_rel_path": "Contents/MacOS/Suspicious Package",
            "execution_time": 1651589744.9446769,
            "logged_in_users": [
                "********"
            ],
            "file_bundle_hash_millis": 221.01902961730957
        }
    ],
etc...
```

Is this expected?  The EventDetailURL is configured with %file_sha% as described on the [configuration page](https://santa.dev/deployment/configuration.html#eventdetailurl). And, in most cases, it uses the file_sha256 value.

Thanks for any help!

**Comment by russellhancox** [comment]
That is expected (but undocumented) behavior - `%file_sha%` will either be the identifier for the executable or the identifier of the bundle containing it, if there is one. We did this so that you could be taken to a page on your sync server that lets you vote on an "app" as a whole unit.

We should improve this by adding a `%file_or_bundle_identifier%` key that has the current behavior and switch %file_sha%` to always be the file's identifier.

**Comment by jamiestrachan-shopify** [comment]
Wow, thanks!
I wasn't expecting a reply so quickly.

**Comment by russellhancox** [comment]
This will be included in the 2022.5 release, which is expected to be released at the end of the month.
